src/coprocessor/dag/batch_executor/table_scan_executor.rs

```diff
@@ -628,12 +641,91 @@ mod tests {
         for ranges in test_ranges {
             for cols in &test_cols {
                 for batch_expect_rows in &test_batch_rows {
-                    test_basic_scan(&store, ranges.clone(), cols, batch_expect_rows);
+                    test_basic_scan(&helper, ranges.clone(), cols, batch_expect_rows);
                 }
             }
         }
     }
 
+    #[test]
+    fn test_execution_summary() {
+        let helper = TableScanTestHelper::new();
+
+        let mut executor = BatchTableScanExecutor::new(
+            ExecSummaryCollectorEnabled::new(1),
+            helper.store(),
+            Arc::new(EvalConfig::default()),
+            helper.columns_info_by_idx(&[0]),
+            vec![helper.whole_table_range()],
+            false,
+        )
+        .unwrap();
+
+        executor.next_batch(1);
+        executor.next_batch(2);
+
+        let mut statistics = BatchExecuteStatistics::new(2, 1);
+        executor.collect_statistics(&mut statistics);
+
+        assert_eq!(statistics.scanned_rows_per_range[0], 3);
+        // 0 is none because our output index is 1
+        assert!(statistics.summary_per_executor[0].is_none());
+        assert_eq!(
+            statistics.summary_per_executor[1]
+                .as_ref()
+                .unwrap()
+                .num_produced_rows,
+            3
+        );
+        assert_eq!(
+            statistics.summary_per_executor[1]
+                .as_ref()
+                .unwrap()
+                .num_iterations,
+            2
+        );
+
+        executor.collect_statistics(&mut statistics);
+
+        assert_eq!(statistics.scanned_rows_per_range[0], 3);
+        assert!(statistics.summary_per_executor[0].is_none());
+        assert_eq!(
+            statistics.summary_per_executor[1]
+                .as_ref()
+                .unwrap()
+                .num_produced_rows,
+            3
+        );
+        assert_eq!(
+            statistics.summary_per_executor[1]
+                .as_ref()
+                .unwrap()
+                .num_iterations,
+            2
+        );
```

Comment:
L690-L705 are asserting the statistics are the same as before? Repeated code exist, is there any way to remove it?

Reply:
Yes, it is exactly asserting the statistics are the same as before. This is the test purpose. When nothing happens, the statistics should be the same as before.